### DISCUSSION TITLE
What is the default timeout?

### DISCUSSION THREAD

**Discussion by kishaningithub** [go-resty/resty#465] [Q&A] [answered]
I see a method here to set the http timeout

https://github.com/go-resty/resty/blob/1792d629c3c6ffcb416497e742e38282a4ab73aa/client.go#L514

but i am unable to find what is the default timeout when we create a client using `resty.New()` ?

**Comment by moorereason** [comment]
The default timeout of 30s is set in [transport.go](https://github.com/go-resty/resty/blob/7a0e1ea9b189eefb4e7ac22da8e2af1d9c1bd623/transport.go).

**Reply by kishaningithub** [reply]
@moorereason The `createTransport()` is not called in the`resty.New()` call sequence 

I also simulated this, steps below

# Simulation steps

1. Start a server which delays the response by say 5 minutes
```shell
$ docker run --env DELAY_RESPONSE=5m -p 8080:8080 kishanb/slow-http:1.0.1
```

2. Simple resty code which consumes the above

```go
package main

import (
	"github.com/go-resty/resty/v2"
	"log"
)

func main() {
	client := resty.New()

	resp, err := client.R().Get("http://localhost:8080")

	if err != nil {
		log.Fatalf(err.Error())
	}

	log.Println(resp)
}
```

## Result

The response from go program prints http response after 5 minutes.

**Comment by jeevatkm** [comment, marked as answer]
@kishaningithub In the flow of method `resty.New()`, the `createTranport(nil)` called at https://github.com/go-resty/resty/blob/1792d629c3c6ffcb416497e742e38282a4ab73aa/client.go#L1048-L1050 if `hc.Transport` is `nil`
